readthedocs/projects/views/private.py

```diff
@@ -527,19 +527,18 @@ def project_notifications(request, project_slug):
         slug=project_slug,
     )
 
-    email_form = EmailHookForm(data=request.POST or None, project=project)
-    webhook_form = WebHookForm(data=request.POST or None, project=project)
+    email_form = EmailHookForm(data=None, project=project)
+    webhook_form = WebHookForm(data=None, project=project)
 
     if request.method == 'POST':
-        if email_form.is_valid():
-            email_form.save()
-        if webhook_form.is_valid():
-            webhook_form.save()
-        project_dashboard = reverse(
-            'projects_notifications',
-            args=[project.slug],
-        )
-        return HttpResponseRedirect(project_dashboard)
+        if 'email' in request.POST.keys():
+            email_form = EmailHookForm(data=request.POST, project=project)
```

Comment:
You can leave the all this code as it was, but calling the `is_valid` method on the right form only after checking `request.POST.keys`.

Reply:
That raises this type of situation
![screenshot from 2018-12-11 19-59-26](https://user-images.githubusercontent.com/29149191/49807248-9955a180-fd7f-11e8-9163-97f8a19b641f.png)

I have submitted empty form for Webhook Notification URL, but it is also showing error for Email Field.

Reply:
This is the flow to me:

```python
if 'url' in request.POST.keys() and webhook_form.is_valid():
    webhook_form.save()
if 'email' in request.POST.keys() and email_form.is_valid():
    email_form.save()
```

The rest of the code should be as it was before.

Reply:
This will also give rise to the same situation (tried on local server).
This is what is happening
```python3
>>> from readthedocs.projects.forms import WebHookForm
>>> from readthedocs.projects.models import Project
>>> proj = Project.objects.get(slug='pikachu-demo')
>>> data = {'url': ''}
>>> form = WebHookForm(data=data, project=proj)
>>> form
<WebHookForm bound=True, valid=Unknown, fields=(url)>
>>> form.errors
{'url': ['This field is required.']}
>>> form
<WebHookForm bound=True, valid=False, fields=(url)>
```

So with these lines
```python3
    email_form = EmailHookForm(data=request.POST or None, project=project)
    webhook_form = WebHookForm(data=request.POST or None, project=project)
```
even the user hasn't submitted the email_form, the error will rise be shown in the templates even if the `is_valid()` method is not run

Reply:
> even the user hasn't submitted the email_form, the error will rise and will be shown in the templates even if the `is_valid()` method is not run

Really?! Wow! I didn't know that and it's an unexpected behavior to me.

Reply:
I think, it's because email form will then accept `data=request.POST`,
but yeahh.. it's unexpected the error is generated with calling `is_valid()` method